src/main/java/uk/gov/hmcts/reform/hmc/helper/CaseHearingRequestMapper.java

```diff
@@ -2,33 +2,40 @@
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.hmc.data.CancellationReasonsEntity;
 import uk.gov.hmcts.reform.hmc.data.CaseCategoriesEntity;
 import uk.gov.hmcts.reform.hmc.data.CaseHearingRequestEntity;
 import uk.gov.hmcts.reform.hmc.data.HearingEntity;
 import uk.gov.hmcts.reform.hmc.model.CaseCategory;
 import uk.gov.hmcts.reform.hmc.model.CaseDetails;
-import uk.gov.hmcts.reform.hmc.model.CreateHearingRequest;
+import uk.gov.hmcts.reform.hmc.model.DeleteHearingRequest;
 import uk.gov.hmcts.reform.hmc.model.HearingDetails;
+import uk.gov.hmcts.reform.hmc.model.HearingRequest;
 
+import java.time.Clock;
+import java.time.LocalDateTime;
 import java.util.List;
 
-import static uk.gov.hmcts.reform.hmc.constants.Constants.VERSION_NUMBER;
-
 @Component
 public class CaseHearingRequestMapper {
 
     private final CaseCategoriesMapper caseCategoriesMapper;
 
+    private final Clock utcClock;
+
     @Autowired
-    public CaseHearingRequestMapper(CaseCategoriesMapper caseCategoriesMapper) {
+    public CaseHearingRequestMapper(CaseCategoriesMapper caseCategoriesMapper,
+                                    Clock utcClock) {
         this.caseCategoriesMapper = caseCategoriesMapper;
+        this.utcClock = utcClock;
     }
 
-    public CaseHearingRequestEntity modelToEntity(CreateHearingRequest createHearingRequest,
-                                                  HearingEntity hearingEntity) {
+    public CaseHearingRequestEntity modelToEntity(HearingRequest hearingRequest,
+                                                  HearingEntity hearingEntity,
+                                                  Integer requestVersion) {
```

Comment:
Can now use this same method for both POST and PUT since we pass in a `HearingEntity` - on POST the passed argument will be a new one, and on PUT it'll be the existing entity.